addons/website_sale_digital/models/product.py

```diff
@@ -59,9 +60,27 @@ def action_open_attachments(self):
             'type': 'ir.actions.act_window',
             'view_mode': 'kanban,form',
             'view_type': 'form',
-            'context': "{'default_res_model': '%s','default_res_id': %d, 'default_product_downloadable': True}" % (self._name, self.id),
+            'context': {'default_res_model': self._name, 'default_res_id': self.id, 'default_product_downloadable': True, 'default_name': self.product_tmpl_id.name},
             'help': """
                 <p class="oe_view_nocontent_create">Click on create to add attachments for this digital product.</p>
                 <p>The attached files are the ones that will be purchased and sent to the customer.</p>
                 """,
         }
+
+    @api.multi
+    def get_digital_attachment(self):
```

Comment:
The whole code here could be simplified, less search, less variables, easier to read.